src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,43 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateTimeFormatter df = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate dateFromNow = LocalDate.parse(dateFrom, df);
+        LocalDate dateToNow = LocalDate.parse(dateTo, df);
+        int[] countSalary = new int [names.length];
+        for (String datum : data) {
+            String[] currentStringInfo = datum.split(" ");
+            for (int j = 0; j < names.length; j++) {
+                LocalDate today = LocalDate.parse(currentStringInfo[0], df);
+                if (names[j].equals(currentStringInfo[1]) && !dateFromNow.isAfter(today)
+                        && !(dateToNow.isBefore(today))) {
+
+                    countSalary[j] += Integer.parseInt(currentStringInfo[2])
```

Comment:
make operands as a variables so it would be easier to read
```suggestion
                    salary = hourlyRate * hoursWorked;
```